### PR TITLE
Add value proposition and example to readme.

### PR DESCRIPTION
**Pull request by kklein**
# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,41 @@
...
 
-MetaLearners for CATE estimation
+MetaLearners for Conditional Average Treatment Effect (CATE) estimation
+
+The library focuses on providing
+
+- Methodologically sound cross-fitting
+- Convenient access to and reuse of base models
+- Consistent APIs across Metalearners
+- Support for multi-class classification and more than binary treatment variants
```
Comment (FrancescMartiEscofetQC) [inline README.md:16]: Nit, multi-class only works for some metalearners, not sure if we should put it as a focus but feel free to ignore

Comment (kklein) [reply README.md]: Oh yes; great catch!

Comment (kklein) [reply README.md]: [ee6c10a](https://github.com/Quantco/metalearners/pull/44/commits/ee6c10ae072dd6219561ec797d95cf4176b5b730)

**Review by FrancescMartiEscofetQC — APPROVED** [review]
LGTM!

**Pull request merged**